db/migrate/20250717003848_create_username_blocks.rb

```diff
@@ -0,0 +1,22 @@
+# frozen_string_literal: true
+
+class CreateUsernameBlocks < ActiveRecord::Migration[8.0]
+  def change
+    create_table :username_blocks do |t|
+      t.string :username, null: false
+      t.string :normalized_username, null: false
+      t.boolean :exact, null: false, default: false
+      t.boolean :allow_with_approval, null: false, default: false
+
+      t.timestamps
+    end
+
+    add_index :username_blocks, 'lower(normalized_username)', unique: true, opclass: :text_pattern_ops, name: 'index_username_blocks_on_username_lower_btree'
```

Comment:
You might also want an index on `username` since we are sorting by it in the controller. Also no need to lower it if we normalize it to be lowercased (though that really isn't a performance concern).